### PR TITLE
Logout event listener to avoid page reloading

### PR DESCRIPTION
**Pull request by sanbornsen**
- An event is broadcasted over logout
    - Listened through out the comonents
    - Update components accordingly on logout
    - Also removing token from the URL after login
            - Need to find a better way to do it, other than just
	      reloading the app to the base url.

### CODE REVIEW

**Comment by sanbornsen** [comment]
This is for the following issues - 
#199 
#198 - Still not the best way

File: src/app/auth/authentication.service.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export class AuthenticationService {
     if ('token' in params) {
       this.authToken = params['token'];
       localStorage.setItem('auth_token', this.authToken);
+      location.href = location.protocol + '//' + location.host;
```
Comment (sanbornsen) [inline src/app/auth/authentication.service.ts]: Did this to get rid of the token from the URL. Need to figure out a better way. Need help!!

Comment (nimishamukherjee) [inline src/app/auth/authentication.service.ts]: The token does not come in the URL but the page refresh is there.

We need to consider this scenario - the user needs to be sent back to the page from where he clicked login. 
If the user is on View Details and clicks Login, after the login is done, the View Details screen needs to be displayed. 
We need to store the route from where login was called, and after successful login, the user needs to be sent back to the same page.

@joshuawilson your thoughts?

**Comment by nimishamukherjee** [comment]
[test]

**Comment by sanbornsen** [comment]
[test]

**Comment by sanbornsen** [comment]
I did nothing and it passed just by running once again.

**Comment by joshuawilson** [comment]
@nimishamukherjee please let me know when you have reviewed this.

File: src/app/work-item/work-item-list-entry/work-item-list-entry.component.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
 import { Router }                                         from '@angular/router';
 
 import { AuthenticationService } from './../../auth/authentication.service';
+import { Broadcaster } from './../../shared/broadcaster.service';
```
Comment (joshuawilson) [inline src/app/work-item/work-item-list-entry/work-item-list-entry.component.ts]: You don't need the ./ in front of the path if you have ../ to begin with.

File: src/app/work-item/work-item-list-entry/work-item-list-entry.component.ts [hunk 3]
```diff
@@ -51,9 +52,14 @@ export class WorkItemListEntryComponent implements OnInit {
   loggedIn: Boolean = false;
 
   constructor(private auth: AuthenticationService,
+              private broadcaster: Broadcaster,
               private router: Router,
               private workItemService: WorkItemService,
               private logger: Logger) {
+                 this.broadcaster.on<string>('logout')
```
Comment (joshuawilson) [inline src/app/work-item/work-item-list-entry/work-item-list-entry.component.ts]: Should this really be done in the constructor. There are several different life cycle methods that you may want to consider, like ngOnInit. You really need to ask yourself when do you want this method called.

Comment (joshuawilson) [inline src/app/work-item/work-item-list-entry/work-item-list-entry.component.ts]: Here is a reference to it in the docs. https://angular.io/docs/ts/latest/guide/lifecycle-hooks.html

File: src/app/work-item/work-item-list-entry/work-item-list-entry.component.ts (hunk 2 above)
Comment (sanbornsen) [inline src/app/work-item/work-item-list-entry/work-item-list-entry.component.ts]: But we agreed on following one standard.

File: src/app/work-item/work-item-list-entry/work-item-list-entry.component.ts (hunk 3 above)
Comment (sanbornsen) [inline src/app/work-item/work-item-list-entry/work-item-list-entry.component.ts]: Yeah, agree. Constructor is not the best place for it. I'll do the changes.

**Comment by nimishamukherjee** [comment]
[test]

**Comment by joshuawilson** [comment]
[test]

**Comment by nimishamukherjee** [comment]
[test]

**Comment by nimishamukherjee** [comment]
[test]

**Comment by joshuawilson** [comment]
[test]

**Comment by joshuawilson** [comment]
@sanbornsen It appears that @nimishamukherjee squashed the branch this PR is based on. You will need to try and rebase it against the updated branch. If you can't get it to rebase properly, you may need to create a new branch based on the github-login branch and copy your work over to it.

**Comment by sanbornsen** [comment]
[test]

**Pull request merged**